<<<<<<< HEAD:Waterfall-Proxy-Patches/0018-Change-IllegalStateException-to-QuietException-and-e.patch
From ea50d368f983aad09ccd5457435c80d7d6790d1b Mon Sep 17 00:00:00 2001
=======
From 8b3e6f968ad9a19466653d9a08b31da13567b6d2 Mon Sep 17 00:00:00 2001
>>>>>>> upstream/master:Waterfall-Proxy-Patches/0017-Change-IllegalStateException-to-QuietException-and-e.patch
From: foss-mc <69294560+foss-mc@users.noreply.github.com>
Date: Wed, 16 Dec 2020 18:40:55 +0800
Subject: [PATCH] Change IllegalStateException to QuietException and explain
 that is a plugin


diff --git a/proxy/src/main/java/net/md_5/bungee/UserConnection.java b/proxy/src/main/java/net/md_5/bungee/UserConnection.java
<<<<<<< HEAD:Waterfall-Proxy-Patches/0018-Change-IllegalStateException-to-QuietException-and-e.patch
index 6d974d74..25fefe81 100644
=======
index fc98b4b1..db79b58a 100644
>>>>>>> upstream/master:Waterfall-Proxy-Patches/0017-Change-IllegalStateException-to-QuietException-and-e.patch
--- a/proxy/src/main/java/net/md_5/bungee/UserConnection.java
+++ b/proxy/src/main/java/net/md_5/bungee/UserConnection.java
@@ -310,7 +310,8 @@ public final class UserConnection implements ProxiedPlayer
 
             if ( getServer() == null && !ch.isClosing() )
             {
-                throw new IllegalStateException( "Cancelled ServerConnectEvent with no server or disconnect." );
+                // FlameCord - Change IllegalStateException to QuietException and explain that is a plugin
+                throw new QuietException("A plugin cancelled ServerConnectEvent with no server or disconnect.");
             }
             return;
         }
-- 
2.32.0.windows.1

